sql/src/main/java/org/apache/druid/sql/calcite/rel/Windowing.java

```diff
@@ -243,8 +244,14 @@ public static Windowing fromCalciteStuff(
       // We know windowProject is a mapping due to the isMapping() check in DruidRules. Check for null anyway,
       // as defensive programming.
       final Mappings.TargetMapping mapping = Preconditions.checkNotNull(
-          partialQuery.getWindowProject().getMapping(),
-          "mapping for windowProject[%s]", partialQuery.getWindowProject()
+              partialQuery.getWindowProject().getProjects().stream().map(RexNode::toString).collect(Collectors.toSet()).size()
+                      < partialQuery.getWindowProject().getProjects().size()
+                      ? Project.getPartialMapping(
+                              partialQuery.getWindowProject().getRowType().getFieldCount(),
+                              partialQuery.getWindowProject().getProjects()
+                      )
+                      : partialQuery.getWindowProject().getMapping(),
+              "mapping for windowProject[%s]", partialQuery.getWindowProject()
```

Comment:
discussed offline; but leave a note here:
I believe we want a real `PartialMapping`  ; you could copy `getPartialMapping` over and change 
`INVERSE_FUNCTION` to `INVERSE_PARTIAL_FUNCTION`

that way we will support partial mapping even in case columns are being discarded...

I also think that `Project.getPartialMapping` might need this change as its apidoc suggersts that [a source might have 0, 1 or more targets.](https://github.com/apache/calcite/blob/b4bcd3b69a39ac40974690d1bd4ea08f364638ce/core/src/main/java/org/apache/calcite/rel/core/Project.java#L420)

Reply:
didn't had to change the mapping type. both `INVERSE_FUNCTION` & `INVERSE_PARTIAL_FUNCTION` are internally using `PartialFunctionImpl`

made use of `source might have 0 targets` by providing the target count as `1 + highest index in our window projection`